File: ros2_ws_p/src/action_package/action_package/count_up_to_client_minimal.py
import rclpy
from rclpy.node import Node
from rclpy.action import ActionClient
from rclpy.action.client import ClientGoalHandle, GoalStatus
from robot_interfaces.action import CountUpTo

class CountUpToClient(Node):
    def __init__(self):
        super().__init__("count_up_to")

def main(args = None):
    rclpy.init(args = args)
    node = CountUpToClient()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_ws_p/src/action_package_cpp/src/count_up_to_client_minimal.cpp
#include "rclcpp/rclcpp.hpp"
#include "rclcpp_action/rclcpp_action.hpp"
#include "robot_interfaces/action/count_up_to.hpp"
#include <functional>

using CountUpTo = robot_interfaces::action::CountUpTo;
using namespace std::placeholders;

class CountUpToClient : public rclcpp::Node
{
    public:
        CountUpToClient() : Node("count_up_to_client")
        {
            RCLCPP_INFO(this->get_logger(), "[CountUpToClient] Node started.");
        }

    private:
        rclcpp_action::Client<CountUpTo>::SharedPtr count_up_to_client_;
};
int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CountUpToClient>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}